### PR TITLE
Hardening Fix-ups

### PR DESCRIPTION
**Pull request by bluemellophone**
Some fix-ups from the Tus, Detection, and ID hardening PRs that have been merged

---

* Remove the hardening hack where preparation is performed in the foreground
* Fix bugs in Sage status display calculation
* Disable hourly Sage sync with Celery (for now)
* Add better sync failure recovery and messaging for Annotations: missing or incorrect annotation bounds
* Add better sync failure recovery and messaging for Assets: file failure to load or send to Sage (e.g., not an image file)
* Add progress tracking percentages for detection in `AssetGroupSighting`
* Add progress tracking percentages for identification in `Sighting`
* Add progress skip messages for identification in `Sighting`
* Move `detection_attempts` back to the relevant section after the Sage request is completed
* Upgrade `BaseProgressSchema` to `DetailedProgressSchema` in `AssetGroupSighting` and `Sighting` schemas, also add more attributes to the Progress schema
* Add the ability to disable `JobControl` task by setting `JOB_CONTROL_CHECK_FREQUENCY=None`

### CODE REVIEW

**Review by bluemellophone — COMMENTED** [review]
LGTM

File: app/modules/sightings/models.py [hunk 1]
```diff
@@ -1420,8 +1455,20 @@ def identified(self, job_id, data):
                 f'Sighting:{self.guid}, Annot:{annot_guid}, algorithm:{algorithm}'
             )
 
+            annotation = Annotation.query.get(annot_guid)
+            if not annotation:
+                raise HoustonException(
+                    log, f'annotation {annot_guid} for {job_id_str} not found'
+                )
+
+            if annotation.progress_identification is not None:
```
Comment (naknomum) [inline app/modules/sightings/models.py:1464]: just curious why you use `... is not None` for these, but above just `if annotation.progress_identification:` ?

**Review by naknomum — APPROVED** [review]
this looks good to me.

File: app/modules/sightings/models.py (hunk 1 above)
Comment (bluemellophone) [reply app/modules/sightings/models.py:1464]: Good catch, will fix

File: app/modules/annotations/models.py [hunk 2]
```diff
@@ -198,12 +207,21 @@ def sync_with_sage(
...
+            )
+            return
+
         sage_request = {
             'image_uuid_list': [to_sage_uuid(self.asset.content_guid)],
             'annot_species_list': [self.ia_class],
             'annot_bbox_list': [self.bounds['rect']],
             'annot_name_list': [annot_name],
-            'annot_theta_list': [self.bounds['theta']],
+            'annot_theta_list': [self.bounds.get('theta', 0)],
```
Comment (naknomum) [inline app/modules/annotations/models.py:224]: thanks for this fix!  just saw this problem happen locally.

File: app/modules/asset_groups/models.py [hunk 3]
```diff
@@ -960,10 +965,21 @@ def send_detection_to_sage(self, model):
                 }
                 # This is necessary because we can only mark self as modified if
```
Comment (hwindsor) [inline app/modules/asset_groups/models.py:966]: The code moved but the comment was left dangling and is not unclear.

**Review by hwindsor — APPROVED** [review]
LGTM, one very minor comment that should be tidied up but isn't a showstopper.

File: app/modules/asset_groups/models.py (hunk 3 above)
Comment (bluemellophone) [reply app/modules/asset_groups/models.py:966]: Removed

**Pull request merged**